doc/source/tutorial/interpolate/2D_fill_from_border.rst

```diff
@@ -0,0 +1,408 @@
+   ...             * coeff_multiplier[:, :, np.newaxis]
+   ...         )
+   ...         coeffs = la.pinvh(
+   ...             poly_edge_values @ poly_edge_values.T,
+   ...             rtol=np.finfo(poly_edge_values.dtype).eps * L * num_coeffs
+   ...         ) @ (poly_edge_values @ border_values)
+   ...         values = polygrid2d(
+   ...             scaled_grid_x[:, 0], scaled_grid_y[0, :],
+   ...             coeffs.reshape(num_coeffs, num_coeffs) * coeff_multiplier[:, :]
+   ...         )
+   ...         plot_and_check_filled_image(values, f"{poly_family:s} degree {num_coeffs - 1:d}")
+   legendre degree 7: Mean error: -3.23e-16 Error std: 0.408
+   chebyshev degree 7: Mean error: 6.81e-17 Error std: 0.408
+   legendre degree 11: Mean error: 3.42e-16 Error std: 0.0241
+   chebyshev degree 11: Mean error: -2.64e-16 Error std: 0.0241
+   legendre degree 15: Mean error: 1.41e-16 Error std: 0.000299
+   chebyshev degree 15: Mean error: 7.43e-17 Error std: 0.000299
+
+
+A different way to restrict the polynomials to avoid interior extrema
+is to insist that the individual polynomials each satisfy the Laplace
+equation, as do, for example,
+:math:`1,x,y,x^2-y^2,xy,x^3-3xy^2,y^3-3yx^2,x^4-6x^2y^2+y^4,x^3y-xy^3,\dots`.
+
+.. plot::
+   :context: close-figs
+
+   >>> def laplace_polynomials(x, y):
+   ...     return np.stack([
+   ...         np.ones_like(x),
+   ...         x,
+   ...         y,
+   ...         x**2 - y**2,
+   ...         x * y,
+   ...         x**3 - 3 * x * y**2,
+   ...         y**3 - 3 * y * x**2,
+   ...         x**4 - 6 * x**2 * y**2 + y**4,
+   ...         x**3 * y - x * y**3,
+   ...         x**5 - 10 * x**3 * y**2 + 5 * x * y**4,
+   ...         y**5 - 10 * y**3 * x**2 + 5 * y * x**4,
+   ...         x**6 - 15 * x**4 * y**2 + 15 * x**2 * y**4 - y**6,
+   ...         3 * x**5 * y - 10 * x**3 * y**3 + 3 * x * y**5,
+   ...         x**7 - 21 * x**5 * y**2 + 35 * x**3 * y**4 - 7 * x * y**6,
+   ...         y**7 - 21 * y**5 * x**2 + 35 * y**3 * x**4 - 7 * y * x**6
+   ...     ])
+   ...
+   >>> poly_edge_values = laplace_polynomials(scaled_edge_x, scaled_edge_y)
+   >>> coeffs = la.pinvh(
+   ...     poly_edge_values @ poly_edge_values.T,
+   ...     rtol=np.finfo(poly_edge_values.dtype).eps * L * num_coeffs
+   ... ) @ (poly_edge_values @ border_values)
+   >>> values = np.tensordot(
+   ...     coeffs, laplace_polynomials(scaled_grid_x, scaled_grid_y), 1
+   ... )
+   >>> plot_and_check_filled_image(values, "Laplace polynomial")
+   Laplace polynomial: Mean error: -3.55e-17 Error std: 0.663
+
+--------------
+Other packages
+--------------
+
+There are packages that will solve Laplace equation with given boundary conditions.
+
+Scikit-Image has an ``inpaint_biharmonic`` function that will solve
+also fill an image from the boundary conditions, in a manner related
+to the Laplace Equation.
+
+
+Statsmodels offers a regression interface that hides the linear
+algebra and offers additional diagnostics.
```

Comment:
Again, would be nice to have a link to a specific statsmodels function(s), and some reasoning to which statsmodels functionality I as a reader should be looking at and why.